front/src/modules/timetableItem/components/Timetable/RoundTrips/TodoColumn.tsx

```diff
@@ -73,6 +73,65 @@ const TodoColumn = ({
     setItemIdToPair(itemId);
   };
 
+  const pairItems = (candidateItem: PairingItem) => {
+    if (!itemIdToPair || !setPairingItems) return;
+    // This item is always in todo
+    const matchingItem = pairingItemsById?.get(itemIdToPair);
+    const timetableItemToPair = timetableItemsWithOpsById.get(itemIdToPair);
+    const candidateItemToPair = timetableItemsWithOpsById.get(candidateItem.id);
+    const candidateTimetableItem = timetableItemsWithOpsById.get(candidateItem.id);
+
+    if (!matchingItem || !timetableItemToPair || !candidateItemToPair || !candidateTimetableItem) {
+      throw new Error('Item to pair not found in todo column');
+    }
+
+    const isValidPair = checkRoundTripCompatible(timetableItemToPair, candidateTimetableItem);
+
+    setPairingItems((prevData) => {
+      const formattedMatchingItem: PairingItem = {
+        ...matchingItem,
+        status: 'roundTrips' as const,
+        pairedItemId: candidateItem.id,
+        isValidPair,
+      };
+
+      const formattedCandidateItem: PairingItem = {
+        ...candidateItem,
+        status: 'roundTrips' as const,
+        pairedItemId: itemIdToPair,
+        isValidPair,
+      };
+
+      // If the candidate item is already paired, we need to move its old pair to todo column
+      let formattedOrphanItem: PairingItem | undefined;
+
+      if (candidateItem.status === 'roundTrips' && pairingItemsById) {
+        const orphanPairingItem = pairingItemsById.get(candidateItem.pairedItemId)!;
+        if (!orphanPairingItem || orphanPairingItem.status !== 'roundTrips')
+          throw new Error('Orphan pairing item not found or wrong status');
+        const { pairedItemId: _, isValidPair: __, ...orphanItem } = orphanPairingItem;
+        formattedOrphanItem = { ...orphanItem, status: 'todo' as const };
+      }
+
+      const updatedPairingItems: PairingItem[] = [
+        formattedMatchingItem,
+        formattedCandidateItem,
+        ...prevData.filter(
+          (item) =>
+            item.id !== formattedOrphanItem?.id &&
+            item.id !== formattedMatchingItem.id &&
+            item.id !== formattedCandidateItem.id
+        ),
+      ];
+
+      return formattedOrphanItem
+        ? [formattedOrphanItem, ...updatedPairingItems]
+        : updatedPairingItems;
```

Comment:
nit (I find it clearer and more concise)
```suggestion
      return [
        ...compact([orphanItem]),
        {
          ...itemA,
          status: 'roundTrips' as const,
          pairedItemId: itemB.id,
          isValidPair,
        },
        {
          ...itemB,
          status: 'roundTrips' as const,
          pairedItemId: itemIdToPair,
          isValidPair,
        },
        ...prevData.filter(
          (item) => item.id !== orphanItem?.id && item.id !== itemA.id && item.id !== itemB.id
        ),
      ];
```

Reply:
Hmm not a fan of the `compact` to avoid the condition, I find it less clear. 
I would prefer to keep it like that if it's ok for you :)

Reply:
Done for the not using formatted items but I kept the last condition with orphans